add combined transformers (#393)
trpc/trpc: www/docs/server/data-transformers.md

});
```

## Different transformers for upload and download

If a transformer should only be used for one directon or different transformers should be used for upload and download (e.g. for performance reasons), you can provide individual transformers for upload and download. Make sure you use the same combined transformer everywhere.

> ### Howto

Reviewer: ```suggestion
### How-to
```

> For demonstration purpose [superjson](https://github.com/blitz-js/superjson) should be used for uploading and faster but insecure [devalue](https://github.com/Rich-Harris/devalue) for downloading data.

Reviewer: ```suggestion
Here [superjson](https://github.com/blitz-js/superjson) is be used for uploading and [devalue](https://github.com/Rich-Harris/devalue) for downloading data, because devalue is a lot faster but insecure to use on the server. 
```

#### 0. Install

```bash
yarn add superjson devalue
```

#### 1. Add to `utils/trpc.ts`

```ts
import superjson from 'superjson';
import devalue from 'devalue';

// [...]

export const transformer = {
  input: superjson,
  output: {
    serialize: (object) => devalue(object),
    deserialize: (object) => eval(`(${object})`),
  },
};

> ```

Reviewer: The only thing I don't like with this is that `devalue` isn't actually needed at all on the client and becomes an unnecessary import - got any idea to work around this?

Reviewer: I guess on the client you only need `input.serialize` & `output.deserialize` and the inverse for the server.... 🤔

Author: At least when using devalue this is no problem. Devalue has no dependencies and should be very small.

Reviewer: ```tsx
export type CombinedDataTransformerClient = {
  input: Pick<DataTransformer, 'serialize'>;
  output: Pick<DataTransformer, 'deserialize'>;
};
```

Reviewer: The types (on the trpcclient) should still reflect that you can omit it if you want so it's up to the dev to optimize to the level they want

Author: I've added `CombinedDataTransformerClient` type and used in client packages, but I am still not sure if this will prevent devalue to be imported on client.

Author: > The types (on the trpcclient) should still reflect that you can omit it if you want so it's up to the dev to optimize to the level they want

Ah, I see. The type is added in https://github.com/trpc/trpc/pull/393/commits/9845d012a7e7bef1f1f77b591a72f61ea85b20ea

#### 2. Add to `createTRPCCLient()`

```ts

> import { transformer } from 'utils/trpc.ts';

Reviewer: ```suggestion
import { transformer } from '../../utils/trpc';
```

> import { transformer } from 'utils/trpc.ts';

Reviewer: ```suggestion
import { transformer } from '../utils/trpc';
```

// [...]

export const client = createTRPCClient<AppRouter>({
  // [...]
  transformer: transformer,
});
```

#### 3. Add to API handler

```ts

> import { transformer } from 'utils/trpc.ts';

Reviewer: ```suggestion
import { transformer } from '../../utils/trpc';
```
